projects/packages/blaze/src/class-blaze.php

```diff
@@ -189,16 +199,27 @@ public static function site_supports_blaze( $blog_id ) {
 	 * @return bool
 	 */
 	public static function should_initialize() {
-		$should_initialize = true;
-		$is_wpcom          = defined( 'IS_WPCOM' ) && IS_WPCOM;
-		$connection        = new Jetpack_Connection();
-		$site_id           = Jetpack_Connection::get_site_id();
+		$is_wpcom   = defined( 'IS_WPCOM' ) && IS_WPCOM;
+		$connection = new Jetpack_Connection();
+		$site_id    = Jetpack_Connection::get_site_id();
 
 		// Only admins should be able to Blaze posts on a site.
 		if ( ! current_user_can( 'manage_options' ) ) {
 			return false;
 		}
 
+		// Allow short-circuiting the Blaze initialization via a filter.
+		if ( has_filter( 'jetpack_blaze_enabled' ) ) {
```

Comment:
Is it intentional that we prioritized the filter over the Connection and Sync checks below? I would expect that even if the filter is used, we should still return `false` if there are connection issues or Sync is disabled.

Reply:
In practice, the filter is really used in 2 scenarios:

- In tests
- In third-party plugins / themes' `functions.php` files as a way to force Blaze to false.

With that in mind, I think it's okay to prioritize the filter, since in real life I don't see folks use it to force Blaze to true?